engine/access/rest/websockets/handler.go

```diff
@@ -10,11 +9,13 @@ import (
 	"github.com/onflow/flow-go/engine/access/rest/common"
 	dp "github.com/onflow/flow-go/engine/access/rest/websockets/data_providers"
 	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/irrecoverable"
 )
 
 type Handler struct {
 	*common.HttpHandler
 
+	ctx                 irrecoverable.SignalerContext
```

Comment:
let's add some context for the next person
```suggestion
	// ctx holds the irrecoverable context used to start the REST server
	// typically we do not store contexts within a struct. it is necessary in this case
	// because we need to pass an irrecoverable context into the API backend logic to
	// handle exceptions, and we cannot use the request's context since the websocket
	// connection lives longer than the request duration.
	ctx                 irrecoverable.SignalerContext
```